vm/systemSmartContracts/auction.go

```diff
@@ -1247,18 +1247,20 @@ func (s *stakingAuctionSC) unStakeTokens(args *vmcommon.ContractCallInput) vmcom
 
 	maxValUnstake := s.computeMaxValueThatCanBeUnStaked(registrationData)
 	unStakeValue := big.NewInt(0).SetBytes(args.Arguments[0])
-	//unstakeValueIsOk will be true if unstake value is larger than the one provided in the config OR is exactly the
-	// value that was left to be unstaked
 	unstakeValueIsOk := unStakeValue.Cmp(s.minUnstakeTokensValue) >= 0 || unStakeValue.Cmp(maxValUnstake) == 0
 	if !unstakeValueIsOk {
 		s.eei.AddReturnMessage("can not unstake the provided value either because is under the minimum threshold or " +
 			"is not the value left to be unStaked")
 		return vmcommon.UserError
 	}
+	if unStakeValue.Cmp(maxValUnstake) > 0 {
+		s.eei.AddReturnMessage("can not unstake a larger value than the possible allowed value which is " + unStakeValue.String())
+		return vmcommon.UserError
+	}
 
 	registrationData.TotalStakeValue.Sub(registrationData.TotalStakeValue, unStakeValue)
```

Comment:
Something is still unclear for me. If you decrease TotalStakeValue here by unStakeValue and also you increase TotalUnstaked with unStakeValue in line 1267, how would be correct the return of method computeMaxValueThatCanBeUnStaked ?
For example, if I have TotalStakeValue = 100 and I will unstake 50 => TotalStakeValue will be 50 (100 - 50) after execution of line 1261 and also TotalUnstaked will be 50 (0 + 50) after execution of line 1267. This means that at the next call, the method computeMaxValueThatCanBeUnStaked will return 0 even if I still have 50 TotalUnstaked there. I am missing something?

Reply:
right, remained from old implementation. Fixed & added unit test. Thanks